shell/platform/darwin/ios/framework/Source/FlutterUndoManagerPlugin.mm

```diff
@@ -0,0 +1,118 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/ios/framework/Source/FlutterUndoManagerPlugin.h"
+#import "flutter/shell/platform/darwin/ios/framework/Source/FlutterEngine_Internal.h"
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+
+#include "flutter/fml/logging.h"
+
+#pragma mark - UndoManager channel method names.
+static NSString* const kSetUndoStateMethod = @"UndoManager.setUndoState";
+
+#pragma mark - Undo State field names
+static NSString* const kCanUndo = @"canUndo";
+static NSString* const kCanRedo = @"canRedo";
+
+@implementation FlutterUndoManagerPlugin {
+  id<FlutterUndoManagerDelegate> _undoManagerDelegate;
+}
+
+- (instancetype)initWithDelegate:(id<FlutterUndoManagerDelegate>)undoManagerDelegate {
+  self = [super init];
+
+  if (self) {
+    // `_undoManagerDelegate` is a weak reference because it should retain FlutterUndoManagerPlugin.
+    _undoManagerDelegate = undoManagerDelegate;
+  }
+
+  return self;
+}
+
+- (void)dealloc {
+  [self resetUndoManager];
+  [super dealloc];
+}
+
+- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
+  NSString* method = call.method;
+  id args = call.arguments;
+  if ([method isEqualToString:kSetUndoStateMethod]) {
+    [self setUndoState:args];
+    result(nil);
+  } else {
+    result(FlutterMethodNotImplemented);
+  }
+}
+
+- (NSUndoManager*)undoManager {
+  return _viewController.undoManager;
+}
+
+- (void)resetUndoManager API_AVAILABLE(ios(9.0)) {
+  [self.undoManager removeAllActionsWithTarget:self];
```

Comment:
nits:
I recommend against using dot notation for method invocations. Although It is legal in Objective-C, it decrease the readability of the code. 
Dot is usually used to access properties in Objective-C, this confuses code readers as there's no property named `undoManager`.

Reply:
Gotcha. I think I used to have a property for it (to allow overriding it in tests), so this was just left over from that. Thanks.